R/string2factor.R

```diff
@@ -12,7 +12,22 @@
 #' @template step-return
 #' @family dummy variable and encoding steps
 #' @export
-#' @details If `levels` is given, `step_string2factor` will
+#' @details
+#'  In most cases, if you are planning to use `step_string2factor()`
```

Comment:
Can you add a top-level statement that tells people to only use this in special cases and, instead, set the levels before using any tidymodels functions? That's the message and it is buried in a lot of minutiae.

Reply:
Do you mean in the description? That seems like a good addition to me to what is already in the details.